tensorrt_llm/llmapi/llm_args.py

```diff
@@ -1407,25 +1380,47 @@ class TrtLlmArgs(BaseLlmArgs):
             default=None, description="Extended runtime perf knob config.")
 
     calib_config: Optional[CalibConfig] = Field(
-        default=None, description="Calibration config.")
+        default_factory=CalibConfig, description="Calibration config.")
 
     embedding_parallel_mode: str = Field(
         default='SHARDING_ALONG_VOCAB',
         description="The embedding parallel mode.")
 
     fast_build: bool = Field(default=False, description="Enable fast build.")
 
+    # BuildConfig is introduced to give users a familiar interface to configure the model building.
+    build_config: Optional[object] = Field(
+        default=None,
+        description="Build config.",
+        json_schema_extra={"type": f"Optional[{get_type_repr(BuildConfig)}]"})
+
     # Private attributes
     _auto_parallel_config: Optional[AutoParallelConfig] = PrivateAttr(
         default=None)
     # This is used to hold the options for convert_checkpoint
     _convert_checkpoint_options: Dict[str,
                                       Any] = PrivateAttr(default_factory=dict)
 
+    # This state helps to manage the build_config initialization.
+    _build_config_init_method: Optional[Literal["from_kwargs", "from_engine",
```

Comment:
Couldn't this also be handled by a `model_validator` that checks if an engine directory was specified?

Reply:
Removed this as they become unnecessary after breaking down the logic into a sequental model_validators, thanks for the great suggestions!